python/paddle/incubate/hapi/model.py

```diff
@@ -1554,27 +1558,102 @@ def save_inference_model(self,
 
             model.save_inference_model('inference_model')
         """
-        assert not fluid.in_dygraph_mode(
-        ), 'Save inference model must in static mode!'
 
-        prog = self._adapter._progs.get('test', None)
-        assert prog, \
-            "Model is not ready, please call `model.prepare()` first"
+        def get_inout_spec(all_vars, return_name=False):
+            result_list = []
+            valid_var_dict = {}
+            valid_vars = [var for var in all_vars if isinstance(var, Variable)]
+            for var in valid_vars:
+                valid_var_dict[var.name] = var
+            result_list = valid_vars
+            if return_name:
+                result_list = [var.name for var in result_list]
 
-        infer_prog = prog.clone(for_test=True)
+            return result_list
 
-        input_names = [v.name for v in self._adapter._input_vars['test']]
-        endpoints = self._adapter._endpoints['test']['output']
+        if fluid.in_dygraph_mode():
+            layer = self.network
+            fluid.disable_dygraph()
+
+            # 1. input check
+            prog_translator = ProgramTranslator()
+            if not prog_translator.enable:
```

Comment:
`prog_translator.enable` is a method, it will always return `True`， please use `prog_translator.enable_declarative`

Reply:
Thanks.